src/xinterpreter.cpp

```diff
@@ -74,10 +79,9 @@ namespace xpyt
         // Monkey patching "from IPython import get_ipython"
         sys.attr("modules")["IPython.core.getipython"] = get_kernel_module();
 
-        // Monkey patching "import linecache". This monkey patch does not work with Python2.
-#if PY_MAJOR_VERSION >= 3
-        sys.attr("modules")["linecache"] = get_linecache_module();
-#endif
+
+        // add get_ipython to global namespace
+        exec(py::str("from IPython.core.getipython import get_ipython"));
```

Comment:
You could also do:

```python
py::globals()["get_ipython"] = get_kernel_module();
```

I fear that the `exec` could end up importing IPython if not executed after the above line.

Reply:
good point, but in this case we need to be careful. `get_kernel_module` should be called only once, because we should only create one instance of InteractiveShell. We might fix that by implementing `InteractiveShell` as a singleton, but it's not yet done.

Reply:
Actually it can be called as many times as you want, it will create the `InteractiveShell` only once. This is ensured by the `static` at this line: https://github.com/jupyter-xeus/xeus-python/blob/ec7338cafbe0f172bf9f1dcd0c0088713b35d167/src/xcomm.cpp#L226 that will create the module instance only at the first time the function is called. Next times it will return the already created module.

Reply:
oh, nice. I didn't know you could use `static` with variables.

Reply:
Yep :) It's really nice: https://www.studytonight.com/cpp/static-keyword.php